### PR TITLE
Fix compiler errors when optimizations are disabled

### PR DESCRIPTION
**Pull request by balaji-nordic**

### CODE REVIEW

**Review by jtguggedal — APPROVED** [review]

**Review by jayteemo — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]

File: applications/asset_tracker_v2/src/modules/cloud_module.c [hunk 1]
```diff
@@ -100,9 +100,18 @@ static int connect_retries;
 static struct cloud_data_cfg copy_cfg;
 const k_tid_t cloud_module_thread;
 
-/* Register message IDs that are used with the QoS library. */
-QOS_MESSAGE_TYPES_REGISTER(GENERIC, BATCH, UI, NEIGHBOR_CELLS, WIFI_ACCESS_POINTS,
```
[the remark this answers is not in the document]
Comment (balaji-nordic) [reply applications/asset_tracker_v2/src/modules/cloud_module.c:104]: Yes. I will remove this.

File: applications/asset_tracker_v2/src/modules/cloud_module.c [hunk 2]
```diff
@@ -100,9 +100,18 @@ static int connect_retries;
...
-			   AGPS_REQUEST, PGPS_REQUEST, CONFIG, MEMFAULT);
+/* Message IDs that are used with the QoS library. */
+enum {
+	GENERIC = 0,
```
Comment (simensrostad) [inline applications/asset_tracker_v2/src/modules/cloud_module.c:105]: Is it nessecary to explicitly set `= 0` here? Isn't the first element always 0.

**Review by simensrostad — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -442,6 +442,10 @@ Other libraries
   * Separated the library from the :ref:`nrf53_audio_app` and moved it to :file:`lib/contin_array`.
     Updated code and documentation accordingly.
 
+* :ref:`QoS` library:
+
+  * Removed the :c:macro:`QOS_MESSAGE_TYPES_REGISTER`.
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:447]:
```suggestion
  * Removed the ``QOS_MESSAGE_TYPES_REGISTER`` macro.
```

**Review by divipillai — APPROVED** [review]

File: applications/asset_tracker_v2/src/modules/cloud_module.c (hunk 2 above)
Comment (balaji-nordic) [reply applications/asset_tracker_v2/src/modules/cloud_module.c:105]: Just due to habit and the personal preference for being explicit.

**Pull request merged**